calculator.rb

```diff
@@ -0,0 +1,91 @@
+puts "Welcome to the Nifty Calculator program!"
+
+expression = ["add(+)", "subtract(-)", "multiply(*)", "divide(/)"]
+
+expression.count.times do |i|
+puts "#{i + 1}." + "#{expression[i]}"
+end
+
+prompt = "Which operator would you like to use: "
+operator_error = "Incorrect operator"
+number_error =  "Error - Numbers Only - Try Again"
+
+def calculate_with_addition( num_one, num_two )
+return num_one + num_two
+end
+
+
+def calculate_with_subtraction( num_one, num_two )
+return num_one - num_two
+end
+
+
+def calculate_with_multiplication( num_one, num_two )
+return num_one * num_two
+end
+
+
+def calculate_with_division( num_one, num_two )
+return num_one / num_two
+end
+####################
+in_progress = true
+
+while in_progress
```

Comment:
I really like the variable `in_progress` that is a boolean `true` or `false`. It reads really well as a human to say, "while `in_progress` is true, this loop should execute".